assembly/src/parsers/ast/mod.rs

```diff
@@ -305,3 +297,11 @@ fn sort_procs_by_index(proc_map: &LocalProcMap) -> BTreeMap<u16, ProcedureAst> {
 
     result
 }
+
+/// Sort a map of procedures into a vec, respecting the order set in the map
+fn sort_procs_into_vec(proc_map: LocalProcMap) -> Vec<ProcedureAst> {
```

Comment:
We already have a procedure sorting (the one above called `sort_procs_by_index()`) - any reason not to use that one? Or, if this one is better, we should probably delete the other one.